indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -198,6 +197,30 @@ public void addLock(final String taskid, final TaskLock taskLock)
   }
 
   @Override
+  public void upgradeLock(String taskid, TaskLock taskLock)
```

Comment:
hmm, sure, I don't see why not -- unless the logic becomes too messy, in which case two methods is fine.